### PR TITLE
Setup command functionality

### PR DESCRIPTION
**Pull request by andrewv517**
This PR adds a framework for adding commands to the plugin. You'll notice the `ICommand` interface - this is similar, but different, than the `CommandExecutor` interface. The interface I created is since the only command for our plugin is `/hub`. All commands we want to add will be additions to that - `/hub join`, `/hub leave`, etc. So you'll have a class implementing `ICommand` for both that `join` and `leave` command, instead of making a full on `CommandExecutor`.

### CODE REVIEW

File: src/main/kotlin/com/craftschemers/hub/commands/ICommand.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package com.craftschemers.hub.commands
+
+import com.craftschemers.hub.minigame.Minigame
+import org.bukkit.command.CommandSender
+
+interface ICommand {
+    val name: String
```
Comment (VirtualParticle) [inline src/main/kotlin/com/craftschemers/hub/commands/ICommand.kt]: Nit: blank lines between each line here aren't necessary IMO

File: src/main/kotlin/com/craftschemers/hub/commands/JoinCommand.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    override val name: String = "join",
+    override val aliases: Array<String>? = null,
+    override val canBeConsole: Boolean = false,
+    override val description: String = "Use this command to join a minigame",
+    override val parameters: Array<String>? = null,
+    override val usage: Array<String> = arrayOf("/hub join <Minigame>"),
+    override val permissionMessage: String = "Permission message",
+    override val permission: String = "permission"
+) : ICommand {
+
```
Comment (VirtualParticle) [inline src/main/kotlin/com/craftschemers/hub/commands/JoinCommand.kt]: Do we want to be able to instantiate multiple `JoinCommand`s with different values? If not this whole thing could just be an `object` like this:

```kotlin
object JoinCommand : ICommand {
    override val name = "join"
    override val aliases: Array<String>? = null
    override val canBeConsole = false
    // ...
```

File: src/main/kotlin/com/craftschemers/hub/managers/HubCommandManager.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.craftschemers.hub.commands.ICommand
+import com.craftschemers.hub.commands.JoinCommand
+import com.craftschemers.hub.minigame.Minigame
+import org.bukkit.ChatColor
+import org.bukkit.command.Command
+import org.bukkit.command.CommandExecutor
+import org.bukkit.command.CommandSender
+
+class HubCommandManager : CommandExecutor {
```
Comment (VirtualParticle) [inline src/main/kotlin/com/craftschemers/hub/managers/HubCommandManager.kt]: Same thing as above with `JoinCommand`, unless we want to instantiate multiple instances of this, we can make this all a single `object`.

File: src/main/kotlin/com/craftschemers/hub/managers/HubCommandManager.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            // ...
+        }
+    }
+
+    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
+        if (args.isEmpty()) {
+            sender.sendMessage("Welcome to the ${ChatColor.GOLD}Hub Plugin!")
+            return true
+        }
+        if (!commands.containsKey(args[0])) {
```
Comment (VirtualParticle) [inline src/main/kotlin/com/craftschemers/hub/managers/HubCommandManager.kt]: Nit, could be `!in`
```suggestion
        if (args[0] !in commands) {
```

Comment (VirtualParticle) [reply src/main/kotlin/com/craftschemers/hub/managers/HubCommandManager.kt]: You could also take advantage of the returned value being `null` if it doesn't have it which might make the code a little more terse.

File: src/main/kotlin/com/craftschemers/hub/commands/JoinCommand.kt (hunk 2 above)
Comment (andrewv517) [reply src/main/kotlin/com/craftschemers/hub/commands/JoinCommand.kt:18]: not sure im following.. why would we want multiple `JoinCommand`s?

File: src/main/kotlin/com/craftschemers/hub/managers/HubCommandManager.kt (hunk 3 above)
Comment (andrewv517) [reply src/main/kotlin/com/craftschemers/hub/managers/HubCommandManager.kt:11]: make what a single object?

**Review by VirtualParticle — APPROVED** [review]

**Pull request merged**